splitter/processor.go

```diff
@@ -39,55 +39,75 @@ func createMessage(index int, row []string) Message {
 
 func (p *Processor) Process(r io.Reader) {
 
-		csvR := csv.NewReader(r)
-		var index = 0
-		var errorCount = 0
+	csvR := csv.NewReader(r)
+	var index = 0
+	var batchSize = config.BatchSize
+	var batchNumber = 1
+	var isFinalBatch = false
+
+batchLoop:
+	for { // each batch
+
+		log.Debug("Processing batch number "+strconv.Itoa(batchNumber)+" index: "+strconv.Itoa(index), nil)
+		var msgs []*sarama.ProducerMessage = make([]*sarama.ProducerMessage, batchSize)
+
+	createBatchLoop:
```

Comment:
Instead of using `break createBatchLoop`, we can remove the need for `createBatchLoop` and wrap the code in an `if ... else` clause.